lib/contentful.js

```diff
@@ -142,46 +151,54 @@ module.exports.fetchCampaignIdsWithKeywords = function () {
 module.exports.fetchKeyword = function (keyword) {
   logger.debug(`contentful.fetchKeyword:${keyword}`);
 
-  return exports.fetchSingleEntry({
-    content_type: 'keyword',
-    'fields.keyword': keyword,
-  });
+  const query = exports.getQueryBuilder()
+    .type('keyword')
+    .keyword(keyword)
+    .build();
+  return exports.fetchSingleEntry(query);
+};
+
+module.exports.fetchKeywords = function fetchKeywords() {
+  logger.debug('contentful.fetchKeywords');
+  const query = exports.getQueryBuilder()
+    .type('keyword')
+    // We store keywords for both Thor and Production as entries, so filter by current environment.
+    .env(process.env.NODE_ENV)
+    .build();
+  return exports.getClient().getEntries(query)
+  .then(response => response.items.map(item => exports.formatKeywordFromResponse(item)))
+  .catch(error => exports.contentfulError(error));
 };
 
 module.exports.fetchKeywordsForCampaignId = function (campaignId) {
   logger.debug(`contentful.fetchKeywordsForCampaignId:${campaignId}`);
 
-  // TODO: refactor into a function that creates the query.
-  const query = {
-    content_type: 'keyword',
+  const query = exports.getQueryBuilder()
+    .type('keyword')
     // We store keywords for both Thor and Production as entries, so filter by current environment.
-    'fields.environment': process.env.NODE_ENV,
-  };
-  if (campaignId) {
-    query['fields.campaign.sys.contentType.sys.id'] = 'campaign';
-    query['fields.campaign.fields.campaignId'] = campaignId;
-  }
-
+    .env(process.env.NODE_ENV)
+    .custom({
+      'fields.campaign.sys.contentType.sys.id': 'campaign',
+      'fields.campaign.fields.campaignId': campaignId,
```

Comment:
The abstraction is there for 2 main reasons. 1) To be able to write tests that can assert that a query has been built (behavior), not what fields the query has (implementation). 2) To structure the query building like you would do it in mongo for example, through chaining calls that add statements to the query.

I agree that there is always the need to "know" the inner workings of the contentful query fields for more complex queries, but this is true for any query builder I have tried, for every document oriented database driver. This is why I think having a method that allows you to just set a custom (complex) query for when we need to, its an acceptable trade-off.